Documentation/async-disaster-recovery.md

```diff
@@ -0,0 +1,460 @@
+# Failover and Failback in RBD Async Disaster Recovery
+
+[RBD mirroring](https://docs.ceph.com/en/latest/rbd/rbd-mirroring/)
```

Comment:
RBD Mirroring is an extremely complicated feature and books could be written on it. So the goal of our docs in Rook is to make it as concise and clear as possible for users to follow, and provide links to more details if they want to understand more deeply. How we present the docs will really make a difference for whether the feature will be used, so it is important to spend time organizing and clarifying the docs. Since so much time is invested in implementing the feature, we need to invest time in the docs by multiple members of the team, and testing the instructions to make sure everything works end-to-end. 

Some more specific suggestions:
- We really need to break this down into multiple docs, it is just too complicated for a single doc. One doc would be an overview, then perhaps separate docs for initial configuration of the clusters, configuration of pools, then a doc for steps to failover/failback. We may want to make the overview page a top-level topic (similar to [Ceph Tools](https://rook.github.io/docs/rook/v1.7/ceph-tools.html)) with multiple subtopics).
- The docs need an overview section at the top of the doc that explain what the user will accomplish in that doc
- Longer docs could use a table of contents with links to each section in the doc, see [this example](https://rook.github.io/docs/rook/v1.7/ceph-common-issues.html#table-of-contents-)

At the end of the day, a non-rbd-replication-expert (including me) should be able to follow the docs and get everything working. I do best with simple features, so let's see how to simplify as much as possible. :)

Reply:
Sure, breaking it down into multiple docs, with each doc covering certain aspects seems good to me. As we can then take the liberty of being descriptive at places as we have multiple docs with each doc focused on a part.

1. Initial doc can shed some light on the Overview.
2. Second doc can cover the Disaster recovery description and how we can set up DR with rook.
3. Lastly, in the third doc, we can cover all the failover and failback steps for Planned migration and DR use cases.

Thoughts?

Reply:
Yes sounds good. For 2 and 3 we could also consider multiple docs for each one if needed, but maybe not. Let's see how it looks with just the 3 docs to start.

Reply:
Thanks for the suggestion. Now we have 3 separate docs for the same. One added benefit of the overview doc is that, once we have a doc for failover and failback steps for Metro DR, we can link it in the Overview doc as well :)